### ISSUE TITLE
Users don't get cached & Not able to use app.listen when client is sharded

### ISSUE THREAD

**Issue by avixityyt** [discordjs/discord.js#6187]
**Please describe the problem you are having in as much detail as possible:**
 so i'm trying to use discord.js master with express to make a api which shows whatever i want , lets say client.users.cache.get(myID).avatarURL() , but thing is it doesn't cache , only when i run   the command on Discord it shows , when i do client.users.cache.size it shows 3 out of 180k users , it doesn't make sense , my caching option thing is this
 ```js
 makeCache: Options.cacheEverything()
 ```  below code that is working , but i can't get users which is so weird.
 Also when i'm using sharded manager replacing 'client' it always says this error from app.listen
 ```js
Error: listen EADDRINUSE: address already in use :::1888
```
 
**Include a reproducible code sample here, if possible:**

```js
const createServer = client => {
  const express = require('express')
  const app = express()

  app.get("/stats", (req, res) => {

    const guild = client.guilds.cache.size
    const userCount = client.guilds.cache.reduce((a, b) => a + b.memberCount, 0)
    const cmdCount = client.commands.size
    const avixAv = client.users.cache.get('459025800633647116').avatarURL({ dynamic: true , size: 4096 })

    res.json({
      guilds: guild,
      userCount: userCount,
      cmdCount: cmdCount,
      devAvatar: avixAv,
    })
    
  })

  return app
}
const app = createServer(client)
```

**Further details:**

- discord.js version: `^13.0.0-dev.4886ae2.1627300977`
- Node.js version: v14.17.0
- Operating system: Windows

**Relevant client options:**

- partials: ```['MESSAGE', 'CHANNEL', 'REACTION' , 'GUILD_MEMBER' , 'USER']```
- gateway intents: ```[
      'GUILDS',
    'GUILD_MEMBERS',
    'GUILD_BANS',
    'GUILD_EMOJIS_AND_STICKERS',
    'GUILD_INTEGRATIONS',
    'GUILD_WEBHOOKS',
    'GUILD_INVITES',
    'GUILD_VOICE_STATES',
    'GUILD_MESSAGES',
    'GUILD_MESSAGE_REACTIONS',
    'GUILD_MESSAGE_TYPING',
    'DIRECT_MESSAGES',
    'DIRECT_MESSAGE_REACTIONS',
    'DIRECT_MESSAGE_TYPING'
  ]```
- other: makeCache: Options.cacheEverything()

**Comment by advaith1** [comment]
users only get cached when the bot gets the user in a Discord gateway event or API response. users cant just get cached out of nowhere; you should fetch them.

**Comment by avixityyt** [comment]
> users only get cached when the bot gets the user in a Discord gateway event or API response. users cant just get cached out of nowhere; you should fetch them.

Yes , you're right , talking about fetching , this **should** cache all users?
```js
client.guilds.cache.forEach(guild => {
  guild.members.cache.forEach(member => {
    if(member.user.bot) return
    console.log(member)
  })
})
```
or by fetching you mean in different way? Maybe i'm doing wrong , but i'm cached only when i used a command , after restart when i check my api respone it doesn't show the avatar link , i do command on discord it shows , pretty confusing , but that's how it's doing for me.

**Comment by advaith1** [comment]
the code you sent doesn't contact Discord at all so it won't really do anything

you can fetch a user with `await client.users.fetch(id)`, [see docs](https://discord.js.org/#/docs/main/master/class/UserManager?scrollTo=fetch)

it's working when you send a message in Discord because discord.js receives the message event from Discord, which includes your user object, and it gets cached from there.

**Comment by avixityyt** [comment]
> the code you sent doesn't contact Discord at all so it won't really do anything
> 
> you can fetch a user with `await client.users.fetch(id)`, [see docs](https://discord.js.org/#/docs/main/master/class/UserManager?scrollTo=fetch)
> 
> it's working when you send a message in Discord because discord.js receives the message event from Discord, which includes your user object, and it gets cached from there.

oh fair enough , but what about not being able to run express when i've set up sharding? cuz without it it works fine , but sharding it says `Error: listen EADDRINUSE: address already in use :::1888` , am i missing something , as i have nothing else running on that port , would appriciate if you/anyone know reason , trying  to figure out for 2+ days :/

**Comment by monbrey** [comment]
You're attempting to run the same server, on the same port, on each shard. They're clashing with each other. Move the server logic out of your bot/client file, and retrieve data from each shard as necessary.
